src/test/groovy/com/mateusz/jakuszko/roomforyou/facade/ReservationDbFacadeSpecification.groovy

```diff
@@ -0,0 +1,181 @@
+    }
+
+    def "should create a reservation - correct reservation dates"() {
+        expect:
+        reservationDbFacade.createReservation(createReservation(startDate, endDate)) != null
+
+        where:
+        startDate                    | endDate
+        LocalDate.now().plusDays(0)  | LocalDate.now().plusDays(0)
+        LocalDate.now().plusDays(0)  | LocalDate.now().plusDays(1)
+        LocalDate.now().plusDays(1)  | LocalDate.now().plusDays(3)
+        LocalDate.now().plusDays(24) | LocalDate.now().plusDays(26)
+        LocalDate.now().plusDays(18) | LocalDate.now().plusDays(20)
+    }
+
+    def "should update a reservation"() {
+        given:
+        ReservationDto reservationDto = reservationDbFacade.getReservation(idsList.get(RESERVATION_INDEX))
+        when:
+        updateReservation(reservationDto)
+        reservationDbFacade.updateReservation(reservationDto)
+        then:
+        verifyAllFields(reservationDto,
+                LocalDate.now().plusDays(22),
+                LocalDate.now().plusDays(26))
+    }
+
+    def "should delete reservation"() {
+        when:
+        reservationDbFacade.deleteReservation(idsList.get(RESERVATION_INDEX))
+        then:
+        reservationDbService.getReservation(idsList.get(RESERVATION_INDEX)).isEmpty()
+    }
+
+
+    private List<Long> prepareData() {
+        Customer customer = Customer.builder()
+                .name("Mateusz")
+                .surname("Jakuszko")
+                .username("matanos")
+                .password("abc123")
+                .role("Admin")
+                .email("mateusz.jakuszko@gmail.com")
+                .build()
+
+        Apartment apartment = Apartment.builder()
+                .city("Terespol")
+                .street("Kraszewskiego")
+                .streetNumber("26")
+                .apartmentNumber(5)
+                .latitude(123.0)
+                .longitude(321.0)
+                .customer(customer)
+                .build()
+
+        Reservation reservation = Reservation.builder()
+                .startDate(LocalDate.now().plusDays(20))
+                .endDate(LocalDate.now().plusDays(24))
+                .apartment(apartment)
+                .customer(customer)
+                .build()
+
+        customerDbService.save(customer, passwordEncoder)
+        apartmentDbService.save(apartment)
+
+        List<Apartment> apartments = [apartment]
+        List<Reservation> reservations = [reservation]
+        apartment.setReservations(reservations)
+
+        customer.setApartments(apartments)
+        customer.setReservations(reservations)
+
+        reservationDbService.save(reservation)
+        apartmentDbService.update(apartment)
+        customerDbService.update(customer)
+
+        List<Long> ids = new ArrayList<>()
+        ids.add(CUSTOMER_INDEX, customer.getId())
+        ids.add(APARTMENT_INDEX, apartment.getId())
+        ids.add(RESERVATION_INDEX, reservation.getId())
+        return ids
```

Comment:
Meh... Skoro mamy tylko trzy "rzeczy" do zarządzania w tym teście to czy nie lepiej stworzyć je niezależnie niż przechowywać w taki sposób.

Jak widzę w teście:
`idsList.get(RESERVATION_INDEX)` to zastanawiam się:
- czym jest idsList?
- skąd się wziął i czym jest RESERVATION_INDEX?

a jak mam gdzieś w given:
`Reservation reservation = createSomeReservation()`
a potem gdzieś w dalej w teście np `repository.findById(reservation.id)` to łatwo prześledzić co jest czym i jak powstało

Reply:
zrobiłem refaktor tej części i muszę przyznać, że faktycznie jest to strasznie nieczytelne. Dzięki za cenne uwagi!